intake_elasticsearch/elasticsearch_seq.py

```diff
@@ -52,19 +52,27 @@ def __init__(self, query, qargs={}, metadata={}, **es_kwargs):
 
         super(ElasticSearchSeqSource, self).__init__(metadata=metadata)
 
-    def _run_query(self, size=None, end=None):
+    def _run_query(self, size=None, end=None, slice_id=None, slice_max=None):
         if size is None:
             size = self._size
         if end is not None:
             size = min(end, size)
+
+        slice_dict = None
+        if slice_id is not None and slice_max is not None:
+            slice_dict = {'slice': {'id': slice_id, 'max': slice_max}}
         try:
             q = json.loads(self._query)
             if 'query' not in q:
                 q = {'query': q}
+
+            if slice_dict:
+                q.update(slice_dict)
             s = self.es.search(body=q, size=size, scroll=self._scroll,
                                **self._qargs)
         except (JSONDecodeError, TypeError):
-            s = self.es.search(q=self._query, size=size, scroll=self._scroll,
+            s = self.es.search(body=slice_dict, q=self._query,
```

Comment:
Can there be no slices if not using a JSON query?
You may not have gathered from the code, but the try: branch is the full query structure, which is written as a JSON string, and the except: branch is the simpler Lucene syntax.

Reply:
I am not sure I get what you mean here. [Documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-uri-request.html) lists the allowed parameters of the URI search (using `q`) and slice is not included.  The slice AFAIK needs to be passed in the payload of the `POST` operation of the search request. method.  That's why I am including the `slice_dict` as value of the `body` parameter.

Reply:
I must admit I don't know the API well - but both partitioned and non-partitioned should be possible for both forms of the search method. It may well be that non-partitioned is exactly the same, but with the number of partitions equal to one, that might be the simplest way to code it.

Reply:
I don't fully understand what you mean but that is the reason why I pass the `slice` payload so that both search methods support the slicing.
ES API is a bit vague although well documented (oxymoron) e.g if you want one partition passing the following payload won't work

```json
{"slice":{"id":0, "max":1}}
```

This would help for a much cleaner implementation IMHO.